packages/runner-ct/src/iframe/iframes.tsx

```diff
@@ -22,134 +21,135 @@ export function getSpecUrl ({ namespace, spec }, prefix = '') {
+    props.eventManager.on('print:selector:elements:to:console', _printSelectorElementsToConsole)
+
+    props.eventManager.start(props.config)
+
+    setDisposers([
+      autorun(() => {
+        autIframe.toggleSelectorPlayground(selectorPlaygroundModel.isEnabled)
+      }),
+      autorun(() => {
+        autIframe.toggleSelectorHighlight(selectorPlaygroundModel.isShowingHighlight)
+      }),
+      autorun(() => {
+        if (props.state.spec) {
+          _run(props.state.spec, props.config)
+        }
+      }),
+    ])
+
+    iframeModel.listen()
+
+    return () => {
+      props.eventManager.notifyRunningSpec(null)
+      props.eventManager.stop()
+      disposers.forEach((dispose) => {
+        dispose()
+      })
+    }
+  }, [])
 
-    this._disposers.push(autorun(() => {
-      this.autIframe.toggleSelectorPlayground(selectorPlaygroundModel.isEnabled)
-    }))
+  useEffect(() => {
+    props.state.callbackAfterUpdate?.()
+  }, [props.state.callbackAfterUpdate])
 
-    this._disposers.push(autorun(() => {
-      this.autIframe.toggleSelectorHighlight(selectorPlaygroundModel.isShowingHighlight)
-    }))
+  const _toggleSnapshotHighlights = (snapshotProps) => {
+    props.state.snapshot.showingHighlights = !props.state.snapshot.showingHighlights
 
-    this.props.eventManager.start(this.props.config)
+    if (props.state.snapshot.showingHighlights) {
+      const snapshot = snapshotProps.snapshots[props.state.snapshot.stateIndex]
 
-    this.iframeModel = new IframeModel({
-      state: this.props.state,
-      restoreDom: this.autIframe.restoreDom,
-      highlightEl: this.autIframe.highlightEl,
-      detachDom: this.autIframe.detachDom,
-      snapshotControls: (snapshotProps) => (
-        <SnapshotControls
-          eventManager={this.props.eventManager}
-          snapshotProps={snapshotProps}
-          state={this.props.state}
-          onToggleHighlights={this._toggleSnapshotHighlights}
-          onStateChange={this._changeSnapshotState}
-        />
-      ),
-    })
+      autIframe.highlightEl(snapshot, snapshotProps)
+    } else {
+      autIframe.removeHighlights()
+    }
+  }
 
-    this.iframeModel.listen()
+  const _changeSnapshotState = (snapshotProps, index) => {
+    const snapshot = snapshotProps.snapshots[index]
 
-    this._disposers.push(autorun(() => {
-      const spec = this.props.state.spec
+    props.state.snapshot.stateIndex = index
+    autIframe.restoreDom(snapshot)
 
-      if (spec) {
-        this._run(spec, config)
-      }
-    }))
+    if (props.state.snapshot.showingHighlights && snapshotProps.$el) {
+      autIframe.highlightEl(snapshot, snapshotProps)
+    } else {
+      autIframe.removeHighlights()
+    }
   }
 
-  @action _setScriptError = (err: string | undefined) => {
-    this.props.state.scriptError = err
+  const _setScriptError = action((err: string | undefined) => {
+    props.state.scriptError = err
```

Comment:
Another mutation of props. Again - just using the code as-is, trying to minimize changes.